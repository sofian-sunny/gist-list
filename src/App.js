import React, { Component } from "react";
import './App.scss'
import { SearchForm, GistItem } from './components'

export default class App extends Component {

  constructor(props) {
    super(props)
    this.state = {
      gistDataItems: null
    }
  }


  onSearchSubmit = async (searchText) => {
    // const apiUrl = 'https://api.github.com/users/anders/gists'
    const apiUrl = `https://api.github.com/users/${searchText}/gists`
    await fetch(apiUrl)
      .then((response) => response.json())
      .then((data) => {
        this.setState({ gistDataItems: data })
      })
  }

  renderGistListItem = () => {
    const { gistDataItems } = this.state
    if (!(gistDataItems && gistDataItems.length)) return null
    return gistDataItems.map(item => {
      return <GistItem item={item} />
    })
  }

  render() {
    return (
      <div className="App">
        <SearchForm onSubmitSearchForm={this.onSearchSubmit}
        placeholderText="Enter git username" />
        {this.renderGistListItem()}
      </div>
    )
  }
}
