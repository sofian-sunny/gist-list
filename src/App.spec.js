import React from 'react'
import { shallow } from 'enzyme'
import App from './App'

const setUp = (props) => {
  const component = shallow(<App {...props} />)
  return component
}

describe('App Component render without error', () => {

  let wrapper = setUp()

  it('App method should return single page app', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("should update gist data on click", () => {
    const mockSuccessResponse = {};
    const mockJsonPromise = Promise.resolve(mockSuccessResponse)
    const mockFetchPromise = Promise.resolve({
      json: () => mockJsonPromise,
    })
    jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise);

    const onSearchSubmit = jest.fn()
    wrapper.instance().onSearchSubmit('sofian-sunny')
  })
})
