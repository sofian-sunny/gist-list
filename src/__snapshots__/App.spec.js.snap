// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component render without error App method should return single page app 1`] = `ShallowWrapper {}`;
